src/mdbExtensionController.ts

```diff
@@ -397,35 +397,71 @@ export default class MDBExtensionController implements vscode.Disposable {
     );
   }
 
-  public openMongoDBShell(): Promise<boolean> {
-    let mdbConnectionString;
-
-    if (this._connectionController) {
-      const activeConnectionModel = this._connectionController
-        .getActiveConnectionModel()
-        ?.getAttributes({ derived: true });
+  isSslConnection(activeConnectionModel: any): boolean {
+    return !!(
+      activeConnectionModel &&
+      activeConnectionModel.driverOptions &&
+      (activeConnectionModel.driverOptions.sslCA ||
+        activeConnectionModel.driverOptions.sslCert ||
+        activeConnectionModel.driverOptions.sslPass)
+    );
+  }
 
-      mdbConnectionString = activeConnectionModel
-        ? activeConnectionModel.driverUrlWithSsh
-        : '';
-    }
+  public openMongoDBShell(): Promise<boolean> {
+    let mdbConnectionString: any;
+    let mdbTlsOptionsString = '';
 
-    if (!mdbConnectionString) {
+    if (!this._connectionController) {
```

Comment:
I think we still want to check if there is an active connection here for this check.
Maybe:
`if (!this._connectionController || !this._connectionController.isCurrentlyConnected()) {`